### ISSUE TITLE
No dynamic DNS updates for reverse zone

### ISSUE THREAD

**Issue by Zwen23** [OpenVPN/openvpn#426]
Hi, when using the data channel offload driver on a windows client it will not send dynamic dns updates for the reverse zone (PTR records). It sends update for the forward zone though.
Using the exact same config and settings but using wintun (disable-dco) or tap driver (dev tap) reverse DNS updates are sent reliable.
Is there something in the dco driver thats blocking or suppressing these reverse dynamic updates?

P.S.: tried setting both checkboxes in the ipv4 DNS-Settings (register dns..., use dns suffix...) with no lock.
also pushing these settings: push "dhcp-option DOMAIN xxx.xxx.xx", push "register-dns"

Any idea ?

**Comment by cron2** [comment]
Just to be sure: this is "windows automatic DNS", so DNS zones hosted at the ADCs?

I can only guess that this is something about interface flags / interface type, and Windows not recognizing "something" if the interface is DCO, while it works fine on TAP or Wintun...

**Comment by Zwen23** [comment]
Yes , right! Windows automatic dns updates for windows clients that are part of an active directory domain. DNS servers are windows Domain Controllers.

 

Von: Gert Doering ***@***.***> 
Gesendet: Mittwoch, 11. Oktober 2023 23:04
An: OpenVPN/openvpn ***@***.***>
Cc: Zwen23 ***@***.***>; Author ***@***.***>
Betreff: Re: [OpenVPN/openvpn] No dynamic DNS updates for reverse zone (Issue #426)

 

Just to be sure: this is "windows automatic DNS", so DNS zones hosted at the ADCs?

I can only guess that this is something about interface flags / interface type, and Windows not recognizing "something" if the interface is DCO, while it works fine on TAP or Wintun...

**Comment by d12fk** [comment]
Closing this one as it went stale. We'll reopen when there's more information provided.
